libpromises/expand.c

```diff
@@ -188,13 +188,21 @@ static void MapIteratorsFromRval(EvalContext *ctx,
 }
 
 static PromiseResult ExpandPromiseAndDo(EvalContext *ctx, PromiseIterator *iterctx,
-                                        PromiseActuator *act_on_promise, void *param)
+                                        PromiseActuator *act_on_promise, void *param,
+                                        bool actuate_ifelse)
 {
     PromiseResult result = PROMISE_RESULT_SKIPPED;
 
+    /* In the case of ifelse() we must always include an extra round of "actuation"
+     * in the while loop below. PromiseIteratorNext() will return false in the case
+     * that there are doubly-unresolved Rvals like $($(missing)).
+     * We can't add an empty wheel because that is skipped as well as noted in
+     * libpromises/iteration.c ShouldAddVariableAsIterationWheel(). */
+    bool ifelse_actuated = ! actuate_ifelse;
```

Comment:
@vpodzime especially this comment above here. This describes my understanding. Might be a bit long but figured I'd start with too much information. :)

Reply:
Yes, now it makes much more sense. Thanks! Just a nitpick: `!actuate_ifelse`